Pages/Post/History.cshtml.cs

```diff
@@ -0,0 +1,36 @@
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using System.Collections.Generic;
+using System.Data;
+
+namespace budoco.Pages.Post
+{
+    public sealed class HistoryModel : PageModel
+    {
+        [FromQuery]
+        public int Id { get; set; }
+
+        public DataRow DataRow { get; set; }
+
+        public void OnGet()
+        {
+            var sql = @"
+                select
+                    *
```

Comment:
There's no reason for us to read the p_before_val and p_after_val twice. Right now I am reading them to produce that green text with the "hover".  You could write your logic so that you get the text from the page that has already been delivered to the browser, rather than making another round trip to the server. What you are doing feels instantaneous when the server and db on your machine, but that click would have a noticable lag in real life.
So.... no, this implementation isn't okay.  Get the text from the DOM where we've already delivered it.

Reply:
Ok.

Reply:
Now works on demand. This probably simplifies the basic query.

Reply:
I don't understand "simplifies the basic query".  My point is there should be *NO* query.  The info that your popup window needs is already in the DOM of the page.  No need for a trip back to the server.

Reply:
Yes. This is what I do.
I wanted to say that when opening a page, we make a request, or it is possible on demand, since the history will not always be read.

Reply:
Done.

Reply:
I don't understand.  When you view an issue, just like with github, viewing that issue means viewing the entire issue.
Same with Budoco, when you view an issue, we get all the posts (comments, emails, history) and display them as a chronological story.  Viewing an issue WITHOUT its posts is not a feature of Budoco, just like it's not a feature of Github. So, that text that your window is displaying, we have already read it from the db and delivered it to the browser. It is in the dom.  There is no reason to go back to the server to read it again.

Reply:
Like pagination.
I agree, it doesn't make sense in this case.